src/timeline/files.rs

```diff
@@ -0,0 +1,90 @@
+use crate::timeline::FFIZER_DATASTORE_DIRNAME;
+use crate::Result;
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+use cid::Cid;
+use multihash_codetable::{Code, MultihashDigest};
+
+const FILES_FILENAME: &str = "files.json";
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub(crate) struct FileMeta {
+    pub key: String,
+    pub hash: Cid,
+}
+
+impl FileMeta {
+    pub fn with_key(self, key: &String) -> Self {
```

Comment:
🔨 **todo**: 
```suggestion
    pub fn with_key(self, key: &str) -> Self {
```
Never use `&String` in a parameter, use `&str`, see https://doc.rust-lang.org/book/ch04-03-slices.html?highlight=deref%20string#string-slices-as-parameters